process.c

```diff
@@ -3011,6 +3012,31 @@ rb_execarg_fail(VALUE execarg_obj, int err, const char *errmsg)
 }
 #endif
 
+/*
+ *  call-seq:
+ *     Process.argv    -> array
+ *
+ *  Returns the list of original command line arguments passed to the
+ *  Ruby executable, while +ARGV+ are the arguments for the Ruby program.
+ *
+ *  Example:
+ *
+ *      $ ruby -e 'p Process.argv'
+ *      ["ruby", "-e", "p Process.argv"]
+ *
+ *  The arguments are encoded using the filesytem encoding.
```

Comment:
I think it should be the locale encoding instead since we basically want the terminal's encoding.

Reply:
I suppose it's debatable. Command line arguments are pretty intimately tied to file system paths, hence why I did it this way. Another option is just BINARY.

Reply:
It should be the same encoding that `ARGV` uses, which I'd think is the locale encoding, but I haven't checked:
```
$ ruby -e 'p ARGV.first.encoding' é
#<Encoding:UTF-8>
```
On Linux/macOS IIRC the filesystem encoding is just the same as the locale encoding for CRuby, but might differ on other platforms.

Reply:
`ARGV` uses `external_str_new_cstr`.

Reply:
Alright, seems fine then (sorry for the very late reply).